### ISSUE TITLE
reboot command silently exits with 0 despite error from sd_bus_call()

### ISSUE THREAD

**Issue by anssih** [systemd/systemd#11546]
**systemd version the issue has been seen with**
master 8499b2c26d (2019-01-24)
(and originally on v237)

**Used distribution**
Fedora 28
(and originally on buildroot 2018.02.3)

**Expected behaviour you didn't see**
Legacy reboot and halt commands should print an error message and have a non-zero exit code if reboot or halt was not actually started due to e.g. a permission error, similar to how `systemctl reboot` does.

E.g. something like this:

    $ sudo -u nobody ./reboot
    Failed to start reboot.target: Access denied
    See system logs and 'systemctl status reboot.target' for details.
    $ echo $?
    4
    $ sudo -u nobody ./systemctl reboot
    Failed to start reboot.target: Access denied
    See system logs and 'systemctl status reboot.target' for details.
    $ echo $?
    4

I realize different behavior may be expected between with these commands, but I think `reboot` should minimally at least produce a non-zero exit code, though an error message would be a plus.

**Unexpected behaviour you saw**

    $ sudo -u nobody ./reboot
    $ echo $?
    0
    $ sudo -u nobody ./systemctl reboot
    Failed to start reboot.target: Access denied
    See system logs and 'systemctl status reboot.target' for details.
    $ echo $?
    4

Even though start_unit_single() returned -EACCES.

**Steps to reproduce the problem**
1. Build systemd with `-Dlogind=false`.
2. Create the reboot symlink: `ln -s systemctl reboot`
3. Try to run `reboot` as an user that has insufficient permissions, e.g. nobody:
`sudo -u nobody ./reboot`
(enter your password to sudo as needed)
4. On my system a polkit password window is shown. Press Cancel.
5. The command exits with exit code 0 with no reboot done.

**What happens in code**
In the non-logind non-force case halt_main() runs start_with_fallback(), which in turns runs start_unit(), which runs start_unit_one().
1. sd_bus_call_method() in start_unit_one() returns -EACCES.
2. The error message is suppressed by `if (arg_action != ACTION_SYSTEMCTL)` in start_unit_one().
3. start_unit() translates -EACCES from start_unit_one() to exit status 4.
4. start_with_fallback() returns 0 if start_unit() returns a positive value, causing a 0 exit code.

**Comment by anssih** [comment]
Until this is resolved properly, I think I'll apply the following locally as a workaround (I have sysv compat disabled via `-Dsysvinit-path=`):

```patch
diff --git a/src/systemctl/systemctl.c b/src/systemctl/systemctl.c
index 63dae2c872..564eddae4c 100644
--- a/src/systemctl/systemctl.c
+++ b/src/systemctl/systemctl.c
@@ -2958,9 +2958,11 @@ static int start_unit_one(
         if (r < 0) {
                 const char *verb;
 
+#if HAVE_SYSV_COMPAT 
                 /* There's always a fallback possible for legacy actions. */
                 if (arg_action != ACTION_SYSTEMCTL)
                         return r;
+#endif
 
                 verb = method_to_verb(method);
 
@@ -8501,9 +8503,11 @@ static int reload_with_fallback(void) {
 }
 
 static int start_with_fallback(void) {
+        int ret;
+
         /* First, try systemd via D-Bus. */
-        if (start_unit(0, NULL, NULL) >= 0)
-                return 0;
+        if ((ret = start_unit(0, NULL, NULL)) >= 0)
+                return ret;
 
         /* Nothing else worked, so let's try /dev/initctl */
         if (talk_initctl() > 0)

```

**Comment by poettering** [comment]
Fix pending in #12014.
